### PR TITLE
Update eclipse setup instructions

### PR DESCRIPTION
**Pull request by dotta**
Thanks to @gkossakowski for helping on setting up my Eclipse environment for Scala development.

review by @gkossakowski

Let me know if you guys want me to squash any of the below commits.

### CODE REVIEW

**Comment by huitseeker** [comment]
apart from comment on the 1st commit, this LGTM

**Comment by dotta** [comment]
> The git-specific )non-regex-related) semantics of **/.cache is that this matches .cache files in all subdirectories. Why should they not be ignored ?

My reason for changing it derives from practical experience: **/.cache didn't work for me. A `git status` was still reporting changes in `.cache` files. The change fixed the problem for me. But if this isn't the right solution, I've no problem removing the specific commit from this PR.

**Comment by retronym** [comment]
You might have an old version of git.

**Comment by dotta** [comment]
@retronym Good point. I'll remove the commit (should I do it right away and force-push, or first wait for the review process to be over?)

**Comment by retronym** [comment]
just force push

**Comment by dotta** [comment]
> just force push

Done!

File: src/eclipse/README.md [hunk 1]
```diff
@@ -61,6 +61,17 @@ consider them unchanged:
...
+If it doesn’t compile
+=====================
+
+The likely reason is that the build path of the imported projects isn’t correct. This can happen for instance
+when the [version.properties](https://github.com/scala/scala/blob/master/versions.properties) file is updated,
+and Eclipse .classpath of the different projects isn’t updated accordingly. The fix is simple, manually inspect
+the build path of each project and make sure the version of the declared dependencies is in sync with the version
+declared in the `version.properties` file. If it isn’t, update it manually and, when done, don’t forget to share
+your changes via a pull request.
+(We are aware this is manual process is cumbersome. If you feel like scripting this process, pull requests are of course welcomed.)
```
Comment (dragos) [inline src/eclipse/README.md]: s/welcomed/welcome

Comment (dotta) [inline src/eclipse/README.md]: Nice catch.

File: src/eclipse/scaladoc/.classpath [hunk 2]
```diff
@@ -6,8 +6,8 @@
 	<classpathentry combineaccessrules="false" kind="src" path="/scala-compiler"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/scala-library"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/partest-extras"/>
-	<classpathentry kind="var" path="M2_REPO/org/scala-lang/modules/scala-xml_2.11.0-M4/1.0-RC3/scala-xml_2.11.0-M4-1.0-RC3.jar"/>
-	<classpathentry kind="var" path="M2_REPO/org/scala-lang/modules/scala-parser-combinators_2.11.0-M4/1.0-RC1/scala-parser-combinators_2.11.0-M4-1.0-RC1.jar"/>
-	<classpathentry kind="var" path="M2_REPO/org/scala-lang/modules/scala-partest_2.11.0-M4/1.0-RC4/scala-partest_2.11.0-M4-1.0-RC4.jar"/>
+	<classpathentry kind="var" path="M2_REPO/org/scala-lang/modules/scala-xml_2.11.0-M5/1.0-RC4/scala-xml_2.11.0-M5-1.0-RC4.jar"/>
+	<classpathentry kind="var" path="M2_REPO/org/scala-lang/modules/scala-parser-combinators_2.11.0-M5/1.0-RC2/scala-parser-combinators_2.11.0-M5-1.0-RC2.jar"/>
+	<classpathentry kind="lib" path="/Users/mirco/.m2/repository/org/scala-lang/modules/scala-partest_2.11.0-M5/1.0-RC5/scala-partest_2.11.0-M5-1.0-RC5.jar"/>
```
Comment (adriaanm) [inline src/eclipse/scaladoc/.classpath]: replace "/Users/mirco/.m2/repository/" by $M2_REPO ;-)
we should really start generating this file using the versions in versions.properties

**Comment by adriaanm** [comment]
thanks, @dotta!

File: src/eclipse/scaladoc/.classpath (hunk 2 above)
Comment (dotta) [inline src/eclipse/scaladoc/.classpath]: Uuups. Good catch :-)

**Comment by gkossakowski** [comment]
LGTM.

Thanks a lot!

**Comment by dotta** [comment]
Alright, time for some squashing.

**Comment by dotta** [comment]
@gkossakowski I'm done with squashing, please let me know if you want more of it - otherwise, feel free to merge ;)

**Comment by gkossakowski** [comment]
It's all good. LGTM again.

I'll merge without waiting PR validation. There's zero chance this breaks anything.

**Pull request merged**